### PR TITLE
Feature/prescription

### PR DESCRIPTION
**Pull request by MirahSaqib**
Prescriptions functionality completed!
Sequenceid and cancancan added!
Search added!

### CODE REVIEW

File: app/controllers/prescriptions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
+class PrescriptionsController < ApplicationController
+  before_action :nested_resource?
```
Comment (rizwan-munir-7) [inline app/controllers/prescriptions_controller.rb]: why need this?

File: app/controllers/prescriptions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        layout: "pdf.html",
+        orientation: "Landscape",
+        zoom: 1
+      end
+    end
+  end
+
+  # POST /appointments/:appointment_id/prescriptions/add_medicine
+  def add_medicine
+    @item = params[:item]
```
Comment (rizwan-munir-7) [inline app/controllers/prescriptions_controller.rb]: no Need for this action 
can be handled at the client-side

File: app/models/prescription.rb [hunk 3]
```diff
@@ -1,6 +1,38 @@
...
+
+  def update_prescription_medicines(medicines_ids, medicine_quantities)
+    self.prescription_medicines = self.prescription_medicines.build(prescription_medicine_hash(medicines_ids, medicine_quantities))
+    self.save  
+  end
+
+  def prescription_medicine_hash(ids, quantities)
+    medicine_ids_array = ids.split(',')
+    medicines_in_prescription = Array.new
+    medicine_number = 0;
```
Comment (rizwan-munir-7) [inline app/models/prescription.rb]: do we need ' ;  '  here

File: app/views/prescriptions/_prescription_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            <th scope="col">Quantity</th>
+          </tr>
+        </thead>
+        <tbody id="med-table">
+          <% @prescription.prescription_medicines.each do |pres_meds| %>
+            <tr id= <%=pres_meds.medicine_id%> >
+              <th scope="row"><%=pres_meds.medicine_id%></th>
+              <td><%=pres_meds.medicine.name%></td>
+              <td id=<%=pres_meds.medicine.name%>-qty>
+                <%=number_field_tag "med_quantity", params[:med_quantity], min: 1, max: pres_meds.medicine.quantity, value: pres_meds.quantity%>
```
Comment (rizwan-munir-7) [inline app/views/prescriptions/_prescription_form.html.erb]: spaces

File: app/views/prescriptions/_prescription_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            <th scope="col">Name</th>
+            <th scope="col">Quantity</th>
+          </tr>
+        </thead>
+        <tbody id="med-table">
+          <% @prescription.prescription_medicines.each do |pres_meds| %>
+            <tr id= <%=pres_meds.medicine_id%> >
+              <th scope="row"><%=pres_meds.medicine_id%></th>
+              <td><%=pres_meds.medicine.name%></td>
+              <td id=<%=pres_meds.medicine.name%>-qty>
```
Comment (rizwan-munir-7) [inline app/views/prescriptions/_prescription_form.html.erb]: id="<%= pres_meds.medicine.name %>-qty"

File: app/views/prescriptions/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,87 @@
+<% provide(:title, "View Prescription") %>
+<div class="row">
+  <h1 class="simple-heading text-left col-4">Prescription # <%=@prescription.sequence_num%></h1>
+  <div class="col-2"></div>
+  <div class="col-6">
+    <% if request.format == :html %>
```
Comment (rizwan-munir-7) [inline app/views/prescriptions/show.html.erb]: request.format.html?

File: app/views/prescriptions/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        <% end %>
+      <% end %>
+      <%= link_to appointment_prescription_path(@appointment, @prescription, format: :pdf), class: "btn btn-danger", 'data-toggle': 'tooltip', title: 'PDF' do %>
+        <i class="fa fa-file-pdf-o"></i>
+      <% end %>
+    <% end %>
+  </div>
+</div>
+<hr class="horizontal-line-color">
+  <% if request.format == :html %>
```
Comment (rizwan-munir-7) [inline app/views/prescriptions/show.html.erb]: request.format.html?

File: db/migrate/20190804172457_add_sequence_num_to_prescriptions.rb [hunk 8]
```diff
@@ -0,0 +1,26 @@
+class AddSequenceNumToPrescriptions < ActiveRecord::Migration
+  
+  
+  def self.up
+    add_column :prescriptions, :sequence_num, :integer,:null=>false
```
Comment (rizwan-munir-7) [inline db/migrate/20190804172457_add_sequence_num_to_prescriptions.rb]: new hash syntax

File: app/models/prescription.rb [hunk 9]
```diff
@@ -1,6 +1,38 @@
...
+
+  scope :doctor_only, -> (user) { joins(:appointment).where(appointments: { doctor_id: user.id, hospital_id: Hospital.current_id } ) }
+  scope :patient_only, -> (user) { joins(:appointment).where(appointments: { patient_id: user.id, hospital_id: Hospital.current_id } ) }
+
+  def build_prescription_medicines(medicines_ids, medicine_quantities)
+    self.prescription_medicines.build(prescription_medicine_hash(medicines_ids, medicine_quantities))
+    self.save  
+  end
+
+  def update_prescription_medicines(medicines_ids, medicine_quantities)
```
Comment (rizwan-munir-7) [inline app/models/prescription.rb]: both methods doing the same thing except assignment,  we can call build_prescription_medicines method here.

File: app/controllers/passwords_controller.rb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+class PasswordsController < Devise::PasswordsController
+  def create
+    user = User.find_by(email: params[:user][:email])
+    if user && user.hospital.subdomain != request.subdomain
+      flash[:danger] = 'Incorrect email! You do not have an account on this website.'
```
Comment (rizwan-munir-7) [inline app/controllers/passwords_controller.rb]: message can be moved to en.yml

File: app/controllers/prescriptions_controller.rb [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  # POST /appointments/:appointment_id/prescriptions
+  def create
+    respond_to do |format|
+      @prescription.update(created_by: current_user.id)
+      if @prescription.save && @prescription.build_prescription_medicines(params[:prescription][:medicine_tokens], params[:med_quantity])
```
Comment (rizwan-munir-7) [inline app/controllers/prescriptions_controller.rb]: move this outside from respond_to block.

**Comment by rizwan-munir-7** [comment]
cycle -> 1

File: app/controllers/prescriptions_controller.rb (hunk 1 above)
Comment (MirahSaqib) [reply app/controllers/prescriptions_controller.rb]: removed

**Pull request merged**